src/spring/azext_spring/_validators_enterprise.py

```diff
@@ -345,6 +345,23 @@ def _is_valid_app_and_profile_name(pattern):
     return len(parts) == 2 and _is_valid_app_name(parts[0]) and _is_valid_profile_name(parts[1])
 
 
+def validate_acs_create(namespace):
+    if namespace.application_configuration_service_generation is not None:
+        if namespace.enable_application_configuration_service is False:
+            raise ArgumentUsageError("--application_configuration_service_generation can only be set when enable application configuration service.")
+
+
+def validate_acs_create_or_update(namespace):
+    _validate_acs_generation(namespace)
+
+
+def _validate_acs_generation(namespace):
+    if namespace.generation is None:
+        return
+    if (namespace.generation not in list(ConfigurationServiceGeneration)):
+        raise InvalidArgumentValueError("Allowed generation are: " + ', '.join(list(ConfigurationServiceGeneration)))
```

Comment:
After adding `arg_type=get_enum_type()` to the parameter, it is not necessary to valid the input values manually.

Reply:
updated